test/jdk/java/awt/Dialog/ModalExcludedTest.java

```diff
@@ -0,0 +1,315 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Button;
+import java.awt.Dialog;
+import java.awt.FileDialog;
+import java.awt.Frame;
+import java.awt.GridLayout;
+import java.awt.JobAttributes;
+import java.awt.PageAttributes;
+import java.awt.Panel;
+import java.awt.PrintJob;
+import java.awt.TextArea;
+import java.awt.Toolkit;
+import java.awt.Window;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.KeyAdapter;
+import java.awt.event.KeyEvent;
+import java.awt.event.KeyListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.awt.event.WindowListener;
+
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+
+import sun.awt.SunToolkit;
+
+/*
+ * @test
+ * @bug 4813288 4866704
+ * @summary Test for "modal exclusion" functionality
+ * @library /java/awt/regtesthelpers /test/lib
+ * @build PassFailJFrame
+ * @modules java.desktop/sun.awt
+ * @run main/manual ModalExcludedTest
+ */
+
+public class ModalExcludedTest {
+    private static final String INSTRUCTIONS = """
+            1. Press 'Modal dialog w/o modal excluded' button below
+                A window, a modeless dialog and a modal dialog will appear
+                Make sure the frame and the modeless dialog are inaccessible,
+                i.e. receive no mouse and keyboard events. MousePressed and
+                KeyPressed events are logged in the text area below - use it
+                to watch events
+            Close all 3 windows
+
+            2. Press 'Modal dialog w/ modal excluded' button below
+                Again, 3 windows will appear (frame, dialog, modal dialog),
+                but the frame and the dialog would be modal excluded, i.e.
+                behave the same way as there is no modal dialog shown. Verify
+                this by pressing mouse buttons and typing any keys. The
+                RootFrame would be modal blocked - verify this too
```

Comment:
` RootFrame would be modal blocked - verify this too`

How to verify this? Does it mean that we cannot close the RootFrame without closing the modal?

Reply:
yup, the RootFrame is deactivated and can't be interacted with

Reply:
Got it. I was probably checking the "Modal excluded" Frame instead of the RootFrame.